spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/main/java/org/springframework/cloud/function/adapter/gcloud/GcfSpringBootHttpRequestHandler.java

```diff
@@ -59,120 +45,29 @@ public GcfSpringBootHttpRequestHandler(Class<?> configurationClass) {
 		super(configurationClass);
 	}
 
+	/**
+	 * The implementation of a GCF {@link HttpFunction} that will be used as the entrypoint from GCF.
+	 */
 	@Override
 	public void service(HttpRequest httpRequest, HttpResponse httpResponse) throws Exception {
 		Thread.currentThread()
 			.setContextClassLoader(GcfSpringBootHttpRequestHandler.class.getClassLoader());
-		initialize(null);
-
-		Publisher<?> output = apply(extract(convert(httpRequest)));
-		BufferedWriter writer = httpResponse.getWriter();
-		Object result = result(httpRequest, output, httpResponse);
-		if (returnsOutput()) {
-			writer.write(gson.toJson(result));
-			writer.flush();
-		}
-		httpResponse.setStatusCode(200);
-	}
 
-	protected Object convert(HttpRequest event) throws IOException {
-		BufferedReader br = event.getReader();
-		StringBuilder sb = new StringBuilder();
+		initialize(httpRequest);
```

Comment:
I think we should move `initialize()` to the constructor. Also, `httpRequest` doesn't make sense for me here.

Reply:
I checked AWS and Azure, and they both initialize in the request handling method. Why do you think it belongs in the constructor?
For `httpRequest`, I want to get the Spring Team's opinion because the context concept is not well documented.